### PR TITLE
[jit] Add serialization docs to jit/README

### PR DESCRIPTION
**Pull request by driazati**
Documents the serialization format for `torch.jit.save`. Some of the info is copied from @houseroad's internal doc.

[Formatted Markdown](https://github.com/driazati/pytorch/blob/serial_docs/torch/csrc/jit/README.md)

Also refactors the readme to have a heading hierarchy + table of contents

### CODE REVIEW

File: torch/csrc/jit/README.md [hunk 1]
```diff
@@ -13,11 +12,66 @@ For concepts that are actual classes in the JIT, we use capitalized words, e.g.
 
 Sections start with a reference to the source file where the code related to the section resides.
 
-Core Program Representation
----------------------------
-
-
-### Modules ###
+## Table of Contents
```
Comment (suo) [inline torch/csrc/jit/README.md]: Are we going to manually updating this from now on? Or is there a way to auto-update

File: torch/csrc/jit/README.md [hunk 2]
```diff
@@ -1108,12 +1160,207 @@ TODO: differentiation, symbolic autograd,
...
+  if torch.gt(y, 2):
+    x0 = torch.add(x, z, 1)
+  else:
+    x0 = torch.add(x, y, 1)
+  return x0
+```
+
+## Serialization
+
+[export.cpp](export.cpp)
```
Comment (suo) [inline torch/csrc/jit/README.md]: link to torch.proto here I think

File: torch/csrc/jit/README.md [hunk 3]
```diff
@@ -1108,12 +1160,207 @@ TODO: differentiation, symbolic autograd,
...
+        "key": "tensors/0"
+      },
+      "device": "cpu"
+    }
+  ],
+  ...
+}
+```
+
+### `attributes.pkl`
```
Comment (suo) [inline torch/csrc/jit/README.md]: It would be nice to see an example pickle file, similar to how you have examples for model.json

**Review by suo — APPROVED** [review]

File: torch/csrc/jit/README.md (hunk 1 above)
Comment (driazati) [reply torch/csrc/jit/README.md]: Semi-manually is probably the least work, you can throw the whole doc into something like this [https://ecotrust-canada.github.io/markdown-toc/](https://ecotrust-canada.github.io/markdown-toc/) and get the TOC

**Pull request closed without merging**